src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/StoreOrderCallRequestMapper.php

```diff
@@ -0,0 +1,145 @@
+     */
+    public function __construct(
+        IdentificationMapperInterface $identificationMapper,
+        OrderMapperInterface $orderMapper,
+        ArvatoRssRepositoryInterface $repository,
+        ArvatoRssConfig $config
+    ) {
+        $this->identificationMapper = $identificationMapper;
+        $this->orderMapper = $orderMapper;
+        $this->repository = $repository;
+        $this->config = $config;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\ArvatoRssStoreOrderRequestTransfer
+     */
+    public function mapOrderToRequestTransfer(OrderTransfer $orderTransfer)
+    {
+        $requestTransfer = new ArvatoRssStoreOrderRequestTransfer();
+
+        $identification = $this->identificationMapper->map();
+        $identification = $this->updateIdentification($orderTransfer, $identification);
+
+        $order = $this->orderMapper->map(
+            $this->createOrderMapperTransfer($orderTransfer)
+        );
+        $order = $this->mapSpecificParameters($orderTransfer, $order);
+
+        $requestTransfer->setIdentification($identification);
+        $requestTransfer->setOrder($order);
+
+        return $requestTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     * @param \Generated\Shared\Transfer\ArvatoRssOrderTransfer $order
+     *
+     * @return \Generated\Shared\Transfer\ArvatoRssOrderTransfer
+     */
+    protected function mapSpecificParameters(OrderTransfer $orderTransfer, ArvatoRssOrderTransfer $order)
+    {
+        $payment = $orderTransfer->getPayments()[0];
+
+        $order->setPaymentType(
+            $this->config->getPaymentTypeMapping(
+                $payment->getPaymentMethod()
+            )
+        );
+        if ($orderTransfer->getCustomer()) {
+            $order->setDebitorNumber($orderTransfer->getCustomer()->getCustomerReference());
+        }
+
+        return $order;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\OrderMapperTransfer
+     */
+    protected function createOrderMapperTransfer(OrderTransfer $orderTransfer)
+    {
+        $transfer = new OrderMapperTransfer();
+        $customer = $orderTransfer->getCustomer();
+        $transfer->setItems($orderTransfer->getItems());
+        $transfer->setTotals($orderTransfer->getTotals());
+        $transfer->setOrderReference($orderTransfer->getOrderReference());
+        $transfer->setCustomerIsGuest($customer ? $customer->getIsGuest() : true);
+
+        return $transfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     * @param \Generated\Shared\Transfer\ArvatoRssIdentificationRequestTransfer $identification
+     *
+     * @return \Generated\Shared\Transfer\ArvatoRssIdentificationRequestTransfer
+     */
+    protected function updateIdentification(
```

Comment:
IMO we should not get data inside of mapper

Reply:
Moved to reader.